flocker/dockerplugin/_script.py

```diff
@@ -73,7 +75,17 @@ def main(self, reactor, options):
         self._create_listening_directory(PLUGIN_PATH.parent())
 
         # Get the node UUID, and then start up:
-        getting_id = flocker_client.this_node_uuid()
+        # Retry on  *all* errors.
+        getting_id = retry_failure(
+            reactor=reactor,
+            function=flocker_client.this_node_uuid,
+            # Keep trying with +5 second backoff up to a 60 second maximum
+            # delay, with jitter
+            steps=(
+                min(60, i * delay) + uniform(-0.2, 0.2)
+                for i, delay in enumerate(repeat(5), start=1)
```

Comment:
I don't understand why enumerate is useful here, since we have the repeat(). Wouldn't it be clearer to do

    min(60, i * 5) + uniform(-0.2, 0.2) for i in itertools.count(1)

I also have some philosophical issues with infinite retries...at some point, it's better to quit and fail, usually.

It might also be a good idea to add a unit test to test the retry flow? If that's too hard, maybe have a function called get_tests which encapsulate the tests logic, and then this code becomes

    getting_id = retry_failure(reactor=reactor, function=..., steps=get_steps())

and get_steps() would be fairly easy to unit tests. This means that if you ever decide the minimum was different, or you want exponential back-off, you could change get_steps(), edit the unit test to confirm the behavior changed, and be confident.